api/v1alpha1/turbine_types.go

```diff
@@ -0,0 +1,77 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TurbineSpec defines the desired state of Turbine
+type TurbineSpec struct {
+	// Sidecar configuration
+	Sidecar *SidecarConfiguration `json:"sidecar,omitempty"`
+
+	// Hazelcast cluster to connect
+	Hazelcast *HazelcastReference `json:"hazelcast,omitempty"`
+
+	// Pod configuration
+	Pods *PodConfiguration `json:"pods,omitempty"`
+}
+
+type SidecarConfiguration struct {
+	// Name of the Turbine sidecar container
+	// +kubebuilder:default:="turbine-sidecar"
+	Name string `json:"name,omitempty"`
+
+	// Repository of the image used for Turbine sidecar container
+	// +kubebuilder:default:="hazelcast/turbine-sidecar"
+	Repository string `json:"repository,omitempty"`
+
+	// Version of the image used for Turbine sidecar container
+	// +kubebuilder:default:="latest"
+	Version string `json:"version,omitempty"`
+}
+
+type HazelcastReference struct {
+	// Address of the Hazelcast cluster
+	ClusterAddress *string `json:"clusterAddress,omitempty"`
+
+	// Reference to Hazelcast CR
+	Cluster *HazelcastRef `json:"cluster,omitempty"`
```

Comment:
This should be a required field to force users define `Cluster.Name`. We could also remove the pointer since the object will never be empty.

```suggestion
	Cluster HazelcastRef `json:"cluster"`
```